whois-api/src/main/java/net/ripe/db/whois/api/rest/HttpsAPIKeyAuthCustomizer.java

```diff
@@ -0,0 +1,128 @@
+            httpResponse.sendError(HttpStatus.BAD_REQUEST_400, "Bad Request");
+            return;
+        }
+
+        if(!canProceed(httpRequest)) {
+            chain.doFilter(request, response);
+            return;
+        }
+
+        LOGGER.debug("It is a api key request");
+
+        if (RestServiceHelper.isHttpProtocol(httpRequest)){
+            final HttpServletResponse httpResponse = (HttpServletResponse) response;
+            httpResponse.sendError(HttpStatus.UPGRADE_REQUIRED_426, "HTTPS required for Basic authorization");
+            return;
+        }
+
+        try {
+            final String accessKey = ApiKeyUtils.getAccessKey(httpRequest.getHeader(HttpHeaders.AUTHORIZATION));
+
+            final String bearerToken = apiKeyAuthServiceClient.validateApiKey(httpRequest.getHeader(HttpHeaders.AUTHORIZATION), accessKey);
+            chain.doFilter(new HttpApiAuthRequestWrapper((HttpServletRequest) request, accessKey, bearerToken), response);
+
+        } catch (Exception ex) {
+            final HttpServletResponse httpResponse = (HttpServletResponse) response;
+            httpResponse.sendError(HttpStatus.INTERNAL_SERVER_ERROR_500, "Api Key request failed");
+        }
+    }
+
+    private static boolean isValidRequest(HttpServletRequest httpRequest) {
+        return (!StringUtils.isEmpty(httpRequest.getQueryString()) && httpRequest.getQueryString().contains(ApiKeyUtils.APIKEY_ACCESS_QUERY_PARAM)) ||
+                (httpRequest.getHeader(HttpHeaders.AUTHORIZATION) != null && httpRequest.getHeader(HttpHeaders.AUTHORIZATION).startsWith("Bearer"));
+    }
+
+    private boolean canProceed(final HttpServletRequest httpRequest) {
+        if(!isEnabled) {
+            return false;
+        }
+
+       //TODO: Remove this logic when basic auth support is deprecated
+       return ApiKeyUtils.isAPIKeyRequest(httpRequest.getHeader(HttpHeaders.AUTHORIZATION));
+    }
+
+    @Override
+    public void destroy() {
+        // do nothing
+    }
+
+    private static final class HttpApiAuthRequestWrapper extends HttpServletRequestWrapper {
+
+        final private String bearerToken;
+        final private String accessKey;
+
+        private HttpApiAuthRequestWrapper(final HttpServletRequest request,  final String accessKey , final String bearerToken) {
+            super(request);
+            this.bearerToken = bearerToken;
+            this.accessKey = accessKey;
+        }
+
+        @Override
+        public String getQueryString() {
+            final StringBuilder modifiedQueryString = (super.getQueryString() == null) ?
+                    new StringBuilder() :
+                    new StringBuilder(super.getQueryString()).append("&");
+
+            return modifiedQueryString.append(ApiKeyUtils.APIKEY_ACCESS_QUERY_PARAM).append("=").append(accessKey).toString();
+        }
+
+        @Override
+        public String getHeader(final String name) {
+            if(!Objects.equals(name, HttpHeaders.AUTHORIZATION)) {
```

Comment:
Invert logic to make more readable (i.e. if authorization header then return bearer and token)